### PR TITLE
Fix potential race condition between evaluator and partition

### PR DESCRIPTION
**Pull request by lava**
###  :notebook_with_decorative_cover: Description

This came up while pairing on on unrelated PR with @tobim . I have no proof that the described race can actually happen in practice, but intuitively it makes sense to me that it would.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

**Review by tobim — APPROVED** [review]
From looking at the changes, this seems to be correct and I do believe that the issue is real. I think one potential symptom is that a `query_supervisor` would miss a response from the cleaned-up partition and stay in limbo eternally, which in turn would block the corresponding exporter from iterating through the remaining candidate partitions.

**Review by tobim — CHANGES_REQUESTED** [review]
This is missing a changlog entry.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ Every entry has a category for which we use the following visual abbreviations:
 
 ## Unreleased
 
+- 🐞 Removed potential race condition that could lead to a hanging export if
+  a partition was persisted just as it was scanned.
+  [#1295](https://github.com/tenzir/vast/pull/1295)
```
Comment (dominiklohmann) [inline CHANGELOG.md:17]:
```suggestion
- 🐞 A potential race condition that could lead to a hanging export if
  a partition was persisted just as it was scanned no longer exists.
  [#1295](https://github.com/tenzir/vast/pull/1295)
```

**Comment by dominiklohmann** [comment]
@lava can you rebase onto master?

**Comment by lava** [comment]
@dominiklohmann Sorry, didn't notice until after I pushed. Done.

**Review by dominiklohmann — APPROVED** [review]
No functional changes happened since I last reviewed this. CI should catch rebase mistakes.

**Comment by lava** [comment]
Ignoring the mac CI fault since it is https://app.clubhouse.io/tenzir/story/20882/race-condition-in-node-queries-unit-test again.

**Pull request merged**